scripts/update_filetypes_from_github.lua

```diff
@@ -0,0 +1,208 @@
+--                         for this language.
+-- group                 - Name of the parent language. Languages in a group are counted
+--                         in the statistics as the parent language.
+
+local lyaml = require('lyaml')
+
+local Path = require('plenary.path')
+local curl = require('plenary.curl')
+
+local write_file = function(path, string)
+  local fd = assert(vim.loop.fs_open(path, "w", 438))
+  assert(vim.loop.fs_write(fd, string, 0))
+  assert(vim.loop.fs_close(fd))
+end
+
+if FORCE_DOWNLOAD or not Path:new("build/languages.yml"):exists() then
+  local languages_yml = curl.get(
+    'https://raw.githubusercontent.com/github/linguist/master/lib/linguist/languages.yml'
+  ).body
+
+  write_file("build/languages.yml", languages_yml)
+else
+  print("Using already downloaded file!")
+end
+
+local prio = {
+  no_match    = -1,
+  scope       = 1,
+  alias       = 2,
+  exact_match = 3,
+}
+
+local find_filetype = function(name, linguist_info, filetype_set)
+  name = string.lower(name)
+
+  local filetype, priority = nil, -1
+  if filetype_set[name] then
+    filetype, priority = name, prio.exact_match
+  end
+
+  if not filetype then
+    if linguist_info.aliases then
+      for _, ft in ipairs(linguist_info.aliases) do
+        ft = string.lower(ft)
+        if filetype_set[ft] then
+          filetype, priority = ft, prio.alias
+
+          break
+        end
+      end
+    end
+  end
+
+  if not filetype then
+    if linguist_info.tm_scope then
+      local tm_scope_split = vim.split(linguist_info.tm_scope, ".", true)
+      local tm_scope = tm_scope_split[#tm_scope_split]
+
+      if filetype_set[tm_scope] then
+        filetype, priority = tm_scope, prio.scope
+      end
+    end
+  end
+
+  return filetype, priority
+end
+
+local overeager_filetypes = {
+  xml = true,
+}
+
+
+local parse_file = function()
+  local yml_string = Path:new("build/languages.yml"):read()
+  local yml_table = lyaml.load(yml_string)
+
+  local output = {
+    extension = {},
+    file_name = {}
+  }
+  local intervention = {}
+
+  local filetype_completions = vim.fn.getcompletion('', 'filetype')
+
+  local filetype_set = {}
+  for _, completed_ft in ipairs(filetype_completions) do
+    filetype_set[completed_ft] = true
+  end
+
+  local add_extension = function(ext, filetype, priority)
+    -- If we have a better match, don't do it.
+    if output.extension[ext] then
+      if overeager_filetypes[output.extension[ext].filetype] then
+        log.debug("Overager:", output.extension[ext].filetype)
+      elseif output.extension[ext].priority > priority then
+        log.debug(
+          "Skipping:", ext, filetype, priority,
+          "due to existing:", output.extension[ext].priority, output.extension[ext].filetype
+        )
+
+        return
+      else
+        log.debug(
+          "Override:", ext, filetype, priority,
+          "due to existing:", output.extension[ext].priority, output.extension[ext].filetype
+        )
+      end
+    end
```

Comment:
Here was the typo i mentioned. You had `output[ext]` rather than `output.extension[ext]` so that code was not reached